packages/core/core/src/factories/useCartFactory.ts

```diff
@@ -30,6 +29,138 @@ export type UseCartFactoryParams<CART, CART_ITEM, PRODUCT, COUPON> = {
+
+    cart.value = initialState || null;
+
+    const addToCart = async (product: PRODUCT, quantity: number) => {
+      loading.value = true;
+      const updatedCart = await factoryParams.addToCart({
+        currentCart: cart.value,
+        product,
+        quantity
+      });
+      cart.value = updatedCart;
+      loading.value = false;
+    };
+
+    const removeFromCart = async (product: CART_ITEM) => {
+      loading.value = true;
+      const updatedCart = await factoryParams.removeFromCart({
+        currentCart: cart.value,
+        product
+      });
+      cart.value = updatedCart;
+      loading.value = false;
+    };
+
+    const updateQuantity = async (product: CART_ITEM, quantity?: number) => {
+      if (quantity && quantity > 0) {
+        loading.value = true;
+        const updatedCart = await factoryParams.updateQuantity({
+          currentCart: cart.value,
+          product,
+          quantity
+        });
+        cart.value = updatedCart;
+        loading.value = false;
+      }
+    };
+
+    const refreshCart = async () => {
+      loading.value = true;
+      cart.value = await factoryParams.loadCart();
+      saveToInitialState(cart.value);
+      loading.value = false;
+    };
+
+    const clearCart = async () => {
+      loading.value = true;
+      const updatedCart = await factoryParams.clearCart({
+        currentCart: cart.value
+      });
+      cart.value = updatedCart;
+      loading.value = false;
+    };
+
+    const isOnCart = (product: PRODUCT) => {
+      return factoryParams.isOnCart({
+        currentCart: cart.value,
+        product
+      });
+    };
+
+    const applyCoupon = async (coupon: string) => {
+      try {
+        loading.value = true;
+        const { updatedCart, updatedCoupon } = await factoryParams.applyCoupon({
+          currentCart: cart.value,
+          coupon
+        });
+        cart.value = updatedCart;
+        appliedCoupon.value = updatedCoupon;
+      } finally {
+        loading.value = false;
+      }
+    };
+
+    const removeCoupon = async () => {
+      try {
+        loading.value = true;
+        const { updatedCart, updatedCoupon } = await factoryParams.removeCoupon({
+          currentCart: cart.value
+        });
+        cart.value = updatedCart;
+        appliedCoupon.value = updatedCoupon;
+        loading.value = false;
+      } finally {
+        loading.value = false;
+      }
+    };
+
+    // Temporary enabled by default, related rfc: https://github.com/DivanteLtd/next/pull/330
+    onSSR(async () => {
+      if (!cart.value) {
+        await refreshCart();
+      }
+    });
+
+    return {
+      cart: computed(() => cart.value),
+      isOnCart,
+      addToCart,
+      refreshCart,
+      removeFromCart,
+      clearCart,
+      updateQuantity,
+      coupon: computed(() => appliedCoupon.value),
+      applyCoupon,
+      removeCoupon,
+      loading: computed(() => loading.value)
+    };
+  };
+
+  return { useCart, setCart };
+};
+
+/*
```

Comment:
Existence of this commented out code hides the actual diif - would you kindly remove it?